File: rabbit-transform/src/main/java/com/susion/rabbit/transform/core/TransformListener.java
package com.susion.rabbit.transform.core;

import com.susion.rabbit.transform.core.context.TransformContext;

import org.jetbrains.annotations.NotNull;

/**
 * Represents the transformStreamToArray lifecycle listener
 */
public interface TransformListener {

    default void onPreTransform(@NotNull final TransformContext context) {
    }

    default void onPostTransform(@NotNull final TransformContext context) {
    }

}

File: rabbit-gradle-transform/src/main/java/com/susion/rabbit/tracer/transform/core/TransformListener.kt
package com.susion.rabbit.tracer.transform.core

import com.susion.rabbit.tracer.transform.core.context.TransformContext

/**
 * susionwang at 2019-11-13
 */
interface TransformListener {

    fun onPreTransform(context: TransformContext){}

    fun onPostTransform(context: TransformContext){}

}
